op-node/eth/account_proof.go

```diff
@@ -52,7 +52,11 @@ func (res *AccountResult) Verify(stateRoot common.Hash) error {
 		if err != nil {
 			return fmt.Errorf("failed to verify storage value %d with key %s (path %x) in storage trie %s: %w", i, entry.Key, path, res.StorageHash, err)
 		}
-		if !bytes.Equal(val, val) {
+		comparison, err := rlp.EncodeToBytes((*big.Int)(&entry.Value).Bytes())
```

Comment:
I believe there is a `.Big` method on entry.value that would clean up the typecast

Reply:
Nice, this is much cleaner. Fixed.